src/paho/mqtt/client.py

```diff
@@ -490,9 +487,14 @@ def __init__(self, client_id="", clean_session=True, userdata=None,
         if not clean_session and (client_id == "" or client_id is None):
             raise ValueError('A client id must be provided if clean session is False.')
 
-        self._transport = transport
+        if transport.lower() not in ('websockets', 'tcp'):
+            raise ValueError('transport must be "websockets" or "tcp", not %s' % transport)
+        self._transport = transport.lower()
         self._protocol = protocol
-        self._userdata = userdata
+        if isinstance(userdata, collections.Iterable) and not isinstance(userdata, basestring):
+            self._userdata = collections.deque(userdata)
```

Comment:
The client used in the publish module sets the userdata to be a queue. As you say, in the client module, it should be anything.